nulib/dc-nextjs
# Update to populating GTM dataLayer variables; add GTM data to embedded viewer route

## What does this do?
As a path to diagnose the Honeybadger errors stemming from Google Analytics / Google Tag Manager, this PR initially was meant to update how we're adding data to the window `dataLayer` object which Google Tag Manager reads for analytics.   In the course of that work, this PR:
- Updates all major dependencies, including NextJS to `v14`
- Updates how we interact with GTM `dayaLayer` using the `@next/third-parties` package
- Adds GTM data to the embedded viewer page, passing on Work info to GTM reporting
- Removes pre-commit hooks via Husky
- Implements a standard Prettier config, and pushes the code quality check up to the CI via Terraform

## How to test?
Log into our RDC Google Tag Manager Account, and access the `dc.northwestern.edu(nextjs)` container.  Click the Preview button to test:

<img width="1439" alt="image" src="https://github.com/nulib/dc-nextjs/assets/3020266/2ceb37af-1a6f-4a4a-89d7-cda649f00971">

Enter in the URL of the preview branch:  https://preview-4759-gtm-update.d2v1qbdeix3nr2.amplifyapp.com/

Click around in the site on the tracked page, and notice the values of `Data Layer`, and `Variables` on the `VirtualPageView` event.  They should be updating to reflect values on each page.

To check the embedded route firing correctly, open up a local test project and embed a Work from the preview branch app above.  In your test consuming app, copy the URL, and Preview this URL in our GTM account above.  Note the Data Layer and Variables values are pulling through now.
